### PR TITLE
Add an Incident Resolution REST endpoint

### LINKED ISSUE #20085: Migrate the Resolve incident gRPC API to REST
## Description

* The Camunda 8 REST API contains a incident resolution endpoint at `POST /v2/incidents/{incidentKey}/resolution` providing the same functionality as the [gRPC endpoint](https://docs.camunda.io/docs/apis-tools/zeebe-api/gateway-service/#resolveincident-rpc).
* The endpoint is featured in the OpenAPI description as (internally) [described](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit#heading=h.g4017mgfkpt9).

### PR DESCRIPTION
**Pull request by nicpuppa**
## Description

Add an endpoint to the REST API for resolving an incident.

## Related issues

closes #20085

### CODE REVIEW

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1354]: 🙃 You can break this summary up into multiple lines like we do in other attributes in this document, I guess. Functionally, this is fine, of course.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentController.java
Comment (tmetzke) [inline zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentController.java:46]: 🙃 In this case, we might also just do the call directly in the public method since we don't parse, transform, or validate the input further. Just a nitpick though, functionally, this is fine, of course.

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java
Comment (tmetzke) [inline zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:27]: ❓ Since we only have the key here, should we add tests that break the API to ensure we're covered? Providing no key is not really possible (you'll get a 404 but, meh 🤷). We could test with a String like `abc` instead of a number? One thing we should do is test that the incident was not found and we receive an appropriate answer.

**Review by tmetzke — APPROVED** [review]
👍 Really cool, good job, @nicpuppa 🚀 
🔧 I added smaller nitpicks and a test suggestion, nothing that blocks the PR though.

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentController.java
Comment (nicpuppa) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentController.java:46]: Yeah it make sense

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java
Comment (tmetzke) [reply zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:27]: Yeah, I agree on "let's not test the framework" but disagree on "I trust the framework" (personally) 😄

Comment (nicpuppa) [reply zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:27]: Yeah, I shouldn't 🤣

Comment (tmetzke) [reply zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:27]: As long as it's Spring, you shouldn't 😆

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java
Comment (tmetzke) [inline zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java:82]: 🔧 Copy-paste mistake? This exists twice now, if I'm not completely confused 😄

**Review by tmetzke — APPROVED** [review]
🚀 Thanks for applying the suggestions 👍 
🙃 One small copy-paste mistake left, I think, good to go from my end after correcting that.

File: zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java
Comment (nicpuppa) [reply zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/IncidentControllerTest.java]: 🙈

**Pull request merged**
